### PR TITLE
feat: Redirect to portal login page if not logged in

### PR DESCRIPTION
**Pull request by suzubara**
# SC-288

SC-288

## Proposed changes

This PR adds middleware to redirect unauthenticated users to the USSF portal log in page, with a redirect URL parameter which can be used to redirect back to the CMS upon login (this functionality will be completed in https://github.com/USSF-ORBIT/ussf-portal-client/pull/579). This is dependent on a new env var: `PORTAL_URL`

Other changes include:
- Adds a public `/api/sysinfo` endpoint to the CMS app
- Adds the PR template to this repo
- Adds a link to the USSF portal site in the Keystone admin navigation

---

### Test different roles locally:
- Try to access the CMS without being logged in and verify you are redirected to the portal log in page instead of seeing the "You don't have access to this page." message.
- Log into the portal as a user without CMS access (`user1 / user1pass`) and try to navigate to the CMS. You should be redirected to the no access page.
- Log into the portal as a user with CMS access (`cmsuser / cmsuserpass`) and try to navigate to the CMS. You should be able to access it.
- Log out of the CMS and verify you are redirected to the portal log in page (and have been logged out of the portal as well).

## Setup

Testing this PR requires running both the CMS and portal app locally:

1. Make sure you have the portal repo checked out, and the environment variable set to its path (in the CMS repo):
```
export PORTAL_PATH=../ussf-portal-client/
```

2. Start the portal app and other services in docker:
```
yarn portal:up
```

3. Start the CMS app:
```
yarn build && yarn start
```

4. You should be able to access the CMS at http://localhost:3001/ and the portal at http://localhost:3000/

---

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
- [ ] Created/modified automated E2E tests in Cypress
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### CODE REVIEW

**Review by esacteksab — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -30,7 +30,7 @@ COPY --from=builder /app /app
 
 ENV NODE_ENV production
 
-EXPOSE 3000
+EXPOSE 3001
```
Comment (esacteksab) [inline Dockerfile:33]: Is this something you want to land in prod, if so, it'll require quite a bit of infrastructure changes. Can you not just set `PORT=3001` in your `.envrc.local so you don't have the conflict locally?

Comment (suzubara) [reply Dockerfile:33]: no, this change is only to the `e2e` target in the Docker file, so it should only affect the image that is used in Cypress tests

File: keystone.ts [hunk 2]
```diff
@@ -13,13 +15,42 @@ export default withSharedAuth(
       useMigrations: true,
     },
     ui: {
-      /*
+      publicPages: ['/api/sysinfo', '/no-access'],
       pageMiddleware: async ({ context, isValidSession }) => {
-      // TODO - add redirect to portal login here if no session
+        const { req, session } = context
+        const pathname = url.parse(req!.url!).pathname!
```
Comment (abbyoung) [inline keystone.ts:21]: do we need to add the eslint ignore statements throughout like we have in the portal repo, or nah?

File: keystone.ts [hunk 3]
```diff
@@ -13,13 +15,42 @@ export default withSharedAuth(
...
+          // Signed in but no access allowed
+          return { kind: 'redirect', to: '/no-access' }
+        } else if (session) {
+          // Allow access to the no access page :)
+          return
+        }
+
+        // If not other public paths & no session, redirect to login page
+        if (pathname !== '/api/sysinfo' && pathname !== '/no-access') {
+          // CYPRESS DEBUG ONLY
```
Comment (abbyoung) [inline keystone.ts:37]: Is the commented-out block to stay for future cypress debugging?

Comment (abbyoung) [reply keystone.ts:37]: (i realize now this will probably get 🪓 bc ✨ playwright ✨ )

Comment (suzubara) [reply keystone.ts]: yup it's 💀

**Review by abbyoung — APPROVED** [review]
Tested all recommended scenarios locally, works as expected!

**Pull request merged**